transformations/style_paraphraser/transformation.py

```diff
@@ -0,0 +1,251 @@
+            self.device = torch.device(
+                "cuda" if torch.cuda.is_available() else "cpu"
+            )
+        self.args["upper_length"] = upper_length
+        self.args["stop_token"] = "eos" if upper_length == "eos" else None
+        self.args["beam_size"] = beam_size
+        self.args["temperature"] = temperature
+        self.args["top_p"] = top_p
+        self.args["top_k"] = 1
+        self.args["device"] = self.device
+        self.config = BASE_CONFIG
+
+        self.config["global_dense_length"] = 0
+        model = GPT2LMHeadModel.from_pretrained(model_path)
+        model.to(self.device)
+        self.gpt2_model = GPT2ParentModule(gpt2=model, device=device)
+        self.tokenizer = GPT2Tokenizer.from_pretrained(model_path)
+
+    def modify_p(self, top_p):
+        """Set top_p to another value"""
+        self.args["top_p"] = top_p
+
+    def generate(self, sentence, top_p=None, n_samples: int = 1):
+        """
+        Generate paraphrases for a batch of outputs - or for the same but with a top_p != 0.0
+        sentence : str
+            Sentence to paraphrase.
+        top_p : float
+            top_p sampling, between 0.0 and 1.0
+            Default None, resorting to the model's top_p value
+        n_samples : int
+            Number of samples to generate for a sentence.
+            Note: These will be the exact same if you use a greedy sampling (top_p=0.0), so if n_samples > 2, makes sure top_p != 0.0.
+        """
+        sent_text = nltk.sent_tokenize(sentence)
+
+        contexts = [sent_text] * n_samples
+
+        to_ret = []
+        for context_ in contexts:
+            instances = []
+            for context in context_:
+                context_ids = self.tokenizer.convert_tokens_to_ids(
+                    self.tokenizer.tokenize(context)
+                )
+
+                instance = Instance(
+                    self.args,
+                    self.config,
+                    {"sent1_tokens": context_ids, "sent2_tokens": context_ids},
+                )
+                instance.preprocess(self.tokenizer)
+                global_dense_vectors = np.zeros((1, 768), dtype=np.float32)
+                instance.gdv = global_dense_vectors
+                instances.append(instance)
+
+            gpt2_sentences = torch.tensor(
+                [inst.sentence for inst in instances]
+            ).to(self.device)
+            segments = torch.tensor([inst.segment for inst in instances]).to(
+                self.device
+            )
+            init_context_size = instances[0].init_context_size
+            eos_token_id = self.tokenizer.eos_token_id
+
+            generation_length = (
+                None
+                if self.args["stop_token"] == "eos"
+                else len(gpt2_sentences[0]) - init_context_size
+            )
+
+            if self.args["beam_size"] > 1:
+                output = _beam_search(
+                    model=self.gpt2_model.gpt2,
```

Comment:
It seems that only the `GPT2LMHeadModel` class is used for the generation (beam search and sampling). There are only three occurrences of `self.gpt2_model`, except the initialization, the other two occurrences call the huggingface class `model=self.gpt2_model.gpt2`. 

Is the wrapper class `GPT2ParentModule` still needed? I think it is not used (feel free to correct me if I'm misunderstanding the codes).

Reply:
Actually, after further inspection, you are right. That observation passed me in my refactoring. Thank you; I have removed that class now and maintain its functionality.

Reply:
So in other words, fixed :)